scalyr_agent/monitor_utils/k8s.py

```diff
@@ -2142,34 +2147,63 @@ def __init__(self, k8s, port=10255, host_ip=None):
         self._session = requests.Session()
         headers = {
             "Accept": "application/json",
+            "Authorization": "Bearer %s" % k8s.token,
         }
         self._session.headers.update(headers)
 
-        global_log.info("KubeletApi host ip = %s" % host_ip)
-        if host_ip:
-            self._http_host = "http://%s:%d" % (host_ip, port)
-        else:
-            self._http_host = None
+        global_log.info("KubeletApi host ip = %s" % self._host_ip)
+        self._build_host()
         self._timeout = 20.0
 
+    def _build_host(self):
+        if self._host_ip:
+            self._host = "%s://%s:%d" % (self._protocol, self._host_ip, self._port)
+        else:
+            self._host = None
+
+    def _switch_to_fallback(self):
+        self._fallback = True
+        self._port = 10255
+        self._protocol = "http"
+        self._build_host()
+
     def query_api(self, path):
         """ Queries the kubelet API at 'path', and converts OK responses to JSON objects
         """
-        url = self._http_host + path
-        response = self._session.get(url, timeout=self._timeout)
+        url = self._host + path
+        try:
```

Comment:
I'm not a big fan of recursive calls like down below.  I'd rather we do something where this is a loop, and we keep retrying until we exit the loop.  You could just do a `while True` and then clearly always break out of the loop down below unless a local variable named `fallback` is `True`.